client/src/main/java/io/confluent/kafka/schemaregistry/client/security/basicauth/BasicAuthCredentialProviderFactory.java

```diff
@@ -16,28 +16,29 @@
 
 package io.confluent.kafka.schemaregistry.client.security.basicauth;
 
+import java.util.HashMap;
 import java.util.Map;
+import java.util.ServiceLoader;
 
 public class BasicAuthCredentialProviderFactory {
 
-  public static BasicAuthCredentialProvider getBasicAuthCredentialProvider(
-      BasicAuthCredentialSource basicAuthCredentialSource,
-      Map<String, ?> configs) {
+  public static final Map<String, BasicAuthCredentialProvider>
+      basicAuthCredentialProviderMap = new HashMap<>();
 
-    BasicAuthCredentialProvider basicAuthCredentialProvider = null;
-    switch (basicAuthCredentialSource) {
-      case URL:
-        basicAuthCredentialProvider = new UrlBasicAuthCredentialProvider();
-        break;
-      case USER_INFO:
-        basicAuthCredentialProvider = new UserInfoCredentialProvider();
-        break;
-      case SASL_INHERIT:
-        basicAuthCredentialProvider = new SaslBasicAuthCredentialProvider();
-        break;
-      default:
-        break;
+  static {
+    for (BasicAuthCredentialProvider basicAuthCredentialProvider
+        : ServiceLoader.load(BasicAuthCredentialProvider.class)) {
```

Comment:
does this change imply we need to make some documentation changes as well?

Reply:
No, the configuration literals remain the same.

Reply:
Does it? Seems like this is purely an implementation detail, although this new implementation will happen to also find all other `BasicAuthCredentialProvider` implementations that are on the classpath and properly defined via the Service Loader mechanism.

Reply:
yes that's true